### DISCUSSION TITLE
Mailer "from"

### DISCUSSION THREAD

**Discussion by crespire** [janko/rodauth-rails#174] [Q&A] [answered]
How can I change the mailer's `from` field? I've tried to stick: `email_from 'test@test.com'` in the `rodauth_mailer.rb` but that didn't seem to do the trick.

I also thought maybe because it inherits from `ApplicationMailer`, that I would update the default there, but no dice either.

I'm still getting `webmaster@localhost` in my LetterOpener emails. It seems logical that this configuration option should live in `RodauthMailer` but the `Rodauth` `email_from` method didn't seem to do anything.

The other solution I thought of was to manually update the header in the email in the `send_email` block in `rodauth_main.rb`, but that seems very inefficient and generally a bad idea.

**Comment by janko** [comment, marked as answer]
The email from header is set directly from within mailer methods, it's passed to the `mail` method call.

Rodauth's `email_from` field is not used in the mailer, because I found it difficult to set that up reliably at this time.

**Reply by crespire** [reply]
I see. Well it's not a big deal, I suppose. It would be nice to have such a configuration option in the future.

Thank you for the response!

**Reply by janko** [reply]
I mixed something up, for some reason I thought we were talking about email `To` header. If your emails go through the default generated mailer, meaning you have `send_*_email` overridden to call the mailer, then `From` header should be whatever is set in `ApplicationMailer` (the mailer doesn't set it). 

If your emails have `From` header set to `webmaster@localhost`, that means they probably don't go through that mailer, but are created within Rodauth directly. In that case changing `email_from "test@test.com"` in `rodauth_main.rb` should be working.

If you run the Rails demo app, you'll see the `From` header in emails is set to the default from `ApplicationMailer`. So, there is something setup differently in your app.

**Reply by crespire** [reply]
@janko Thank you for the prompt.

You were correct, I did indeed have the mailer commented out. It was an artifact of an earlier iteration of my API where I had decided against sending a verification.

Having restored the mailer, the `email_from` directive works as I expect in `rodauth_main.rb`!

Thank you so much!!
